test/dynamo/test_logging.py

```diff
@@ -0,0 +1,131 @@
+# Owner(s): ["module: dynamo"]
+import functools
+import unittest.mock
+
+import torch
+import torch._dynamo.logging as td_logging
+import torch._dynamo.test_case
+import torch._dynamo.testing
+import torch._logging
+from torch.testing._internal.dynamo_logging_utils import make_test
+
+from torch.testing._internal.inductor_utils import HAS_CUDA
+from torch.testing._internal.logging_utils import LoggingTestCase
+
+requires_cuda = functools.partial(unittest.skipIf, not HAS_CUDA, "requires cuda")
+
+
+def example_fn(a):
+    output = a.mul(torch.ones(1000, 1000))
+    output = output.add(torch.ones(1000, 1000))
+    output.sum().backward()
+    return output
+
+
+def dynamo_error_fn(a):
+    output = a.mul(torch.ones(1000, 1000))
+    output = output.add(torch.ones(10, 10))
+    return output
+
+
+def inductor_error_fn(a):
+    output = torch.round(a)
+    return output
+
+
+def inductor_schedule_fn(a):
+    output = a.add(torch.ones(1000, 1000, device="cuda"))
+    return output
+
+
+ARGS = (torch.ones(1000, 1000, requires_grad=True),)
+
+
+def multi_record_test(name, ty, num_records):
+    @make_test(name)
+    def fn(self, records):
+        fn_opt = torch._dynamo.optimize("inductor")(example_fn)
+        fn_opt(*ARGS)
+        self.assertEqual(len(records), num_records)
+        self.assertIsInstance(records[0].msg, ty)
+
+    return fn
+
+
+def within_range_record_test(name, ty, num_records_lower, num_records_higher):
+    @make_test(name)
+    def fn(self, records):
+        fn_opt = torch._dynamo.optimize("inductor")(example_fn)
+        fn_opt(*ARGS)
+        self.assertGreaterEqual(len(records), num_records_lower)
+        self.assertLessEqual(len(records), num_records_higher)
+        self.assertIsInstance(records[0].msg, ty)
+
+    return fn
+
+
+def single_record_test(name, ty):
+    return multi_record_test(name, ty, 1)
+
+
+class LoggingTests(LoggingTestCase):
+
+    test_bytecode = multi_record_test("bytecode", td_logging.ByteCodeLogRec, 2)
+    test_output_code = multi_record_test("output_code", td_logging.OutputCodeLogRec, 2)
+
+    @requires_cuda()
+    @make_test("schedule")
+    def test_schedule(self, records):
+        fn_opt = torch._dynamo.optimize("inductor")(inductor_schedule_fn)
+        fn_opt(torch.ones(1000, 1000, device="cuda"))
+        self.assertEqual(len(records), 1)
+        self.assertIsInstance(records[0].msg, td_logging.ScheduleLogRec)
```

Comment:
These tests seem brittle. How am I supposed to update them if I modify the logging behavior of any of the components under test?

Reply:
Open to suggestions here, I made some of the tests look for ranges, but for the individual artifacts, I think making sure each of them is logged properly is worth it. If you remove the artifact it's pretty obvious IMO how the test needs to be changed.

Reply:
Discussed offline, I'll add a more descriptive comment.